SDL/EndcapGeometry.cc

```diff
@@ -82,4 +69,9 @@ void SDL::EndcapGeometry<SDL::Dev>::fillGeoMapArraysExplicit(SDL::QueueAcc& queu
   alpaka::wait(queue);
 }
 
-float SDL::EndcapGeometry<SDL::Dev>::getdxdy_slope(unsigned int detid) { return dxdy_slope_[detid]; }
+float SDL::EndcapGeometry<SDL::Dev, true>::getdxdy_slope(unsigned int detid) const {
+  return (detid < dxdy_slope_.size() ? dxdy_slope_.at(detid) : 0);
```

Comment:
This part was reading data out of bounds. We should check why that is happening and how that was affecting performance.

Reply:
Isn't dxdy_slope_ a map? Why does this change make sense?